go/libraries/doltcore/sqle/dsess/dolt_session.go

```diff
@@ -128,12 +129,29 @@ func getPersistedValue(conf config.ReadableConfig, k string) (interface{}, error
 
 	var res interface{}
 	switch value.(type) {
-	case int, int8, int16, int32, int64:
+	case int8:
+		switch strings.ToLower(v) {
+		case "on", "true", "1":
+			return int8(1), nil
+		case "off", "false", "0":
+			return int8(0), nil
+		default:
+			res, err = strconv.ParseInt(v, 10, 8)
+		}
+	case int, int16, int32, int64:
 		res, err = strconv.ParseInt(v, 10, 64)
 	case uint, uint8, uint16, uint32, uint64:
 		res, err = strconv.ParseUint(v, 10, 64)
 	case float32, float64:
 		res, err = strconv.ParseFloat(v, 64)
+	case bool:
```

Comment:
When is this used? All boolean system var should be int8, MySQL doesn't actually have a bool type